src/odoc/fs.ml

```diff
@@ -89,12 +89,25 @@ module Directory = struct
     | Error (`Msg e) -> invalid_arg ("Odoc.Fs.Directory.of_string: " ^ e)
     | Ok p -> Fpath.to_dir_path p
 
-  let ls t =
-    let elts = Sys.readdir (to_string t) |> Array.to_list in
-    List.fold_left elts ~init:[] ~f:(fun acc elt ->
-      let file = File.create ~directory:t ~name:elt in
-      if Fpath.is_file_path file then file :: acc else acc
```

Comment:
If anyone wonders w.r.t. what I'm replacing with that line was absurd. `Fpath` works syntactically it nevers queries the OS.